File: 프로그래머스/unrated/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
def solution(today, terms, privacies):
    answer = []
    
    year, month, day = today.split('.')
    translated_today = ((int(year) * 12) + int(month)) * 28 + int(day)

    periods = {}
    for term in terms:
        type, period = term.split()
        periods[type] = int(period) * 28
    
    for i in range(len(privacies)):
        period, term = privacies[i].split()
        
        target_year, target_month, target_day = period.split('.')
        translated_target = ((int(target_year) * 12) + int(target_month)) * 28 + int(target_day) + periods[term]

        if translated_today >= translated_target:
            answer.append(i+1)
        
    return answer

File: 프로그래머스/unrated/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.java
import java.util.*;

class Solution {
    public int[] solution(String today, String[] terms, String[] privacies) {
        int[] answer;
        
        int totalToday = getTotalDays(today);
        
        HashMap<String, Integer> termsMap = new HashMap<>();
        for (String term : terms) {
            String[] termSplit = term.split(" ");
            termsMap.put(termSplit[0], Integer.parseInt(termSplit[1]) * 28);
        }
        
        List<Integer> answerList = new ArrayList<>();
        for (int i = 0; i < privacies.length; i++) {
            String[] privacy = privacies[i].split(" ");
            int tempTotalDays = getTotalDays(privacy[0]);
            tempTotalDays += termsMap.get(privacy[1]);
            
            if (tempTotalDays <= totalToday) {
                answerList.add(i + 1);
            }
        }
        
        answer = new int[answerList.size()];
        for (int i = 0; i < answerList.size(); i++) {
            answer[i] = answerList.get(i);
        }
        
        return answer;
    }
    
    private int getTotalDays(String today) {
        int totalDays = 0;
        
        String[] todaySplit = today.split("\\.");
        totalDays += Integer.parseInt(todaySplit[0]) * 12 * 28;
        totalDays += Integer.parseInt(todaySplit[1]) * 28;
        totalDays += Integer.parseInt(todaySplit[2]);
        
        return totalDays;
    }
}
